core/lsp/PreemptionTaskManager.cc

```diff
@@ -41,8 +42,20 @@ bool PreemptionTaskManager::tryRunScheduledPreemptionTask() {
         absl::MutexLock lock(&typecheckMutex);
         // Invariant: Typechecking _cannot_ be canceled before or during a preemption task.
         ENFORCE(!epochManager->wasTypecheckingCanceled());
-        preemptTask->run();
+        // The error queue is where typechecking puts all typechecking errors. For a given edit, Sorbet LSP runs
+        // typechecking and then drains the error queue. If we failed to temporarily swap it out during preemption, the
+        // preempted task will see all of the errors that have accumulated thus far on the slow path. Thus, we save the
+        // old error queue and replace so new operation starts fresh
+        auto previousErrorQueue = move(gs.errorQueue);
+        gs.errorQueue = make_shared<core::ErrorQueue>(previousErrorQueue->logger, previousErrorQueue->tracer);
+        gs.errorQueue->ignoreFlushes = true;
```

Comment:
I think I understand your perspective; wouldn't it be good to quickly send out the accumulated errors?

The problem is that typechecking passes only expect to find errors in the files that they explicitly typecheck. In the slow path, that's every file in the workspace (modulo files that have been updated during fast paths that preempt it).  In the fast path, that's all of the files impacted by the change (typically, the files that were edited + files that call methods whose signature changed in those edits). 

At the end of typechecking (on the slow and the fast path), the error queue is drained and errors are grouped per file. If a file was typechecked and has no errors, then the typechecking pass believes that the file is error-free and sends an empty error list to the client to clear out errors from the previous typechecking run.

If you combine those two facts....

If a fast path preempts a slow path, and we let the fast path send out that slow path's accumulated errors, then the slow path will, in turn, believe that all of those files do not contain any errors when it finishes. Thus, it will tell the editor that "all of these files are error free".